include/pistache/common.h

```diff
@@ -52,12 +52,20 @@
 // Until we require C++17 compiler with [[maybe_unused]]
 #define UNUSED(x) (void)(x);
 
+#ifndef MAX_PAYLOAD
+#define MAX_PAYLOAD 4096
+#endif
+
+// Allow compile-time overload
 namespace Pistache {
 namespace Const {
 
-    static constexpr int MaxBacklog = 128;
-    static constexpr int MaxEvents = 1024;
-    static constexpr int MaxBuffer = 4096;
-    static constexpr int ChunkSize = 1024;
+    static constexpr size_t MaxBacklog = 128;
+    static constexpr size_t MaxEvents  = 1024;
+    static constexpr size_t MaxBuffer  = 4096;
+
+    // Defined from CMakeLists.txt in project root
+    static constexpr size_t MaxPayload = MAX_PAYLOAD;
```

Comment:
I was super-excited to test this PR.  When Pistache compiles, "Pistache::Const::MaxPayload" is properly set via the -DPISTACHE_MaX_PAYLOAD.  So the installed .so has this feature.

Unfortunately, the installed header files still have the MaxPayload at 4096 bytes.  When I compile my own project that "#include <pistache.h>" (or equiv), MaxPayload is incorrect and my client-side code makes poor decisions.  We need a way to preserve the compile-time size of MaxPayload so that library users can obtain it, or better yet, an API that can be called that returns whatever value was compiled in (instead of just reading a constant from a header file).

Thoughts?

Reply:
Yup, that seems to be the way to go. I'll get it working.